lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/RestAccessControlModule.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import io.airlift.log.Logger;
+import io.airlift.units.Duration;
+import io.trino.plugin.base.CatalogName;
+import io.trino.spi.connector.ConnectorAccessControl;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Suppliers.memoizeWithExpiration;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.airlift.http.client.HttpClientBinder.httpClientBinder;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class RestAccessControlModule
```

Comment:
Can we have it as a `UriBasedAccessControl`

Reply:
hmm for test you suggested use http so maybe let's keep it consistent and use HttpBasedAccessControl. Also if there for example s3 support in the future and you have something like s3://<s3.location> that might be still considered as uri? WDYT?

Reply:
Yeah !! Sounds good to me